src/engine/plugins/tile.cpp

```diff
@@ -162,8 +220,56 @@ FixedLine coordinatesToTileLine(const util::Coordinate start,
 
     return tile_line;
 }
+
+/**
+ * Converts lon/lat into coordinates inside a Mercator projection tile (x/y pixel values)
+ *
+ * @param point the lon/lat you want the tile coords for
+ * @param tile_bbox the mercator boundaries of the tile
+ * @return a point (x,y) on the tile defined by tile_bbox
+ */
+FixedPoint coordinatesToTilePoint(const util::Coordinate point, const detail::BBox &tile_bbox)
+{
+    const FloatPoint geo_point{static_cast<double>(util::toFloating(point.lon)),
+                               static_cast<double>(util::toFloating(point.lat))};
+
+    const double px_merc = geo_point.x * util::web_mercator::DEGREE_TO_PX;
+    const double py_merc = util::web_mercator::latToY(util::FloatLatitude{geo_point.y}) *
+                           util::web_mercator::DEGREE_TO_PX;
+
+    const auto px = static_cast<std::int32_t>(std::round(
+        ((px_merc - tile_bbox.minx) * util::web_mercator::TILE_SIZE / tile_bbox.width()) *
+        util::vector_tile::EXTENT / util::web_mercator::TILE_SIZE));
+    const auto py = static_cast<std::int32_t>(std::round(
+        ((tile_bbox.maxy - py_merc) * util::web_mercator::TILE_SIZE / tile_bbox.height()) *
+        util::vector_tile::EXTENT / util::web_mercator::TILE_SIZE));
+
+    return FixedPoint{px, py};
 }
 
+/**
+ * Unpacks a single CH edge (NodeID->NodeID) down to the original edges, and returns a list of the
+ * edge data
+ * @param from the node the CH edge starts at
+ * @param to the node the CH edge finishes at
+ * @param unpacked_path the sequence of EdgeData objects along the unpacked path
+ */
+void UnpackEdgeToEdges(const datafacade::BaseDataFacade &facade,
+                       const NodeID from,
+                       const NodeID to,
+                       std::vector<datafacade::BaseDataFacade::EdgeData> &unpacked_path)
+{
+    std::array<NodeID, 2> path{{from, to}};
+    UnpackCHEdge(&facade,
+                 path.begin(),
+                 path.end(),
+                 [&unpacked_path](const std::pair<NodeID, NodeID> & /* edge */,
+                                  const datafacade::BaseDataFacade::EdgeData &data) {
+                     unpacked_path.emplace_back(data);
+                 });
+}
+} // ::detail namespace
```

Comment:
This can be an anonymous namespace to keep the symbols hidden in this TU, right?